### PR TITLE
Add the `kueue.x-k8s.io/managed` label to PodTemplates created by Kueue

### LINKED ISSUE #2856: Add a label to PodTemplates created by Kueue
**What would you like to be added**:

A label or set of labels that are added to every PodTemplate created by Kueue (for the ProvReq mechanism). This should be configurable via the Kueue Configuration API.

**Why is this needed**:

To easily identify Workloads created by Kueue.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by PBundyra**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Adds the `kueue.x-k8s.io/managed` label to PodTemplates created by Kueue

#### Which issue(s) this PR fixes:

Fixes #2856 

#### Does this PR introduce a user-facing change?

```release-note
Label `kueue.x-k8s.io/managed` is now added to PodTemplates created via ProvisioningRequest by Kueue
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
Let's also add it to the ProvisioningRequest objects

File: pkg/controller/admissionchecks/provisioning/controller.go [hunk 1]
```diff
@@ -383,6 +383,9 @@ func (c *Controller) syncProvisionRequestsPodTemplates(ctx context.Context, wl *
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      ptKey.Name,
 					Namespace: ptKey.Namespace,
+					Labels: map[string]string{
+						"kueue.x-k8s.io/managed": "true",
```
[the remark this answers is not in the document]
Comment (PBundyra) [reply pkg/controller/admissionchecks/provisioning/controller.go]: Thanks for pointing this out, done!

File: pkg/constants/constants.go [hunk 2]
```diff
@@ -42,4 +42,6 @@ const (
 	DefaultPendingWorkloadsLimit = 1000
 
 	IsNegativeErrorMsg string = `must be greater than or equal to 0`
+
+	ManagedByKueueLabel = "kueue.x-k8s.io/managed"
```
Comment (alculquicondor) [inline pkg/constants/constants.go:46]: Please use it in the Pod webhook too (and wherever else the same string is used)

Comment (PBundyra) [reply pkg/constants/constants.go:47]: Done

**Review by tenzen-y — COMMENTED** [review]
IIUC, this is only for pv-request, right?

/approve
/assign @alculquicondor for lgtm

**Comment by tenzen-y** [comment]
/release-note-edit
```release-note
Label `kueue.x-k8s.io/managed` is now added to PodTemplates created via ProvisioningRequest by Kueue
```

**Review by alculquicondor — COMMENTED** [review]
/lgtm

IIRC, Kueue doesn't create any other objects.

Well, with the exception of Workloads, which are always managed by Kueue, so a label would be overkill.

**Pull request merged**